### PR TITLE
Test c backtrace with full paths

### PR DESCRIPTION
**Pull request by pitr-ch**
review commit by commit
there are 2 minor jt improvements

### CODE REVIEW

File: tool/jt.rb [hunk 1]
```diff
@@ -1050,85 +1051,98 @@ def test_compiler(*args)
...
+            expected = File.read("#{dir}/expected.txt")
+            unless actual == expected
+              abort "C extension #{dir} didn't work as expected\nActual:\n#{actual}\nExpected:\n#{expected}\n"
+            end
+          else
+            run_ruby "-I#{dir}/lib", "#{dir}/bin/#{gem_name}", out: output_file
+            unless File.read(output_file) == File.read("#{dir}/expected.txt")
+              abort "c extension #{dir} didn't work as expected"
+            end
+          end
```
Comment (chrisseaton) [inline tool/jt.rb]: This seems like quite a lot of logic to go into `jt`. Could this be a file in the `test/truffle` directory?

Comment (pitr-ch) [reply tool/jt.rb]: I just moved what we had into a case to make tests executable selectively. I did not look into any other improvements, but I've also noticed we had quite a lot of functionality here. I may have a look later.

File: tool/jt.rb [hunk 2]
```diff
@@ -1050,85 +1051,98 @@ def test_compiler(*args)
...
         else
-          run_ruby "-I#{dir}/lib", "#{dir}/bin/#{gem_name}", out: output_file
-          unless File.read(output_file) == File.read("#{dir}/expected.txt")
-            abort "c extension #{dir} didn't work as expected"
-          end
+          raise "unknown test: #{test_name}"
         end
       end
     ensure
       File.delete output_file rescue nil
```
Comment (eregon) [inline tool/jt.rb]: Is `output_file` set in this scope? It should be in the `when` above I believe.
And `rescue nil` => `if File.exist?(output_file)` would be better to notice these problems earlier.

File: tool/jt.rb [hunk 3]
```diff
@@ -1050,85 +1051,98 @@ def test_compiler(*args)
...
+          compile_cext gem_name, ext_dir, "#{dir}/lib/#{gem_name}/#{gem_name}.su"
+          case gem_name
+          when 'backtraces'
+            run_ruby "-I#{dir}/lib", "#{dir}/bin/#{gem_name}", err: output_file, continue_on_failure: true
+            actual = File.read(output_file).
+                gsub(TRUFFLERUBY_DIR, '').
+                gsub(/\/cext(_ruby)?\.rb:(\d+)/, '/cext\1.rb:n')
+            expected = File.read("#{dir}/expected.txt")
+            unless actual == expected
+              abort "C extension #{dir} didn't work as expected\nActual:\n#{actual}\nExpected:\n#{expected}\n"
```
Comment (eregon) [inline tool/jt.rb]: No need for `\n` at the end of `abort`, it's like `puts`. But it doesn't hurt either.

**Pull request merged**